src/main/java/de/tum/in/www1/artemis/service/learningpath/LearningPathService.java

```diff
@@ -362,9 +362,12 @@ public LearningPathNavigationOverviewDTO getLearningPathNavigationOverview(long
      */
     public LearningPath findWithCompetenciesAndLearningObjectsAndCompletedUsersById(long learningPathId) {
         LearningPath learningPath = learningPathRepository.findWithCompetenciesAndLectureUnitsAndExercisesByIdElseThrow(learningPathId);
+
+        learningPath.getCompetencies()
+                .forEach(competency -> competency.setExercises(competency.getExercises().stream().filter(Exercise::isVisibleToStudents).collect(Collectors.toSet())));
         // Remove exercise units, since they are already retrieved as exercises
```

Comment:
Please adjust the comment to highlight that this step also filters out unreleased units